diffrax/_solver/srk.py

```diff
@@ -0,0 +1,740 @@
+from dataclasses import dataclass
+from typing import ClassVar, Generic, Optional, TYPE_CHECKING, TypeVar, Union
+from typing_extensions import TypeAlias
+
+import equinox as eqx
+import equinox.internal as eqxi
+import jax
+import jax.lax as lax
+import jax.numpy as jnp
+import jax.tree_util as jtu
+import numpy as np
+from equinox.internal import ω
+from jaxtyping import Array, Float, PyTree
+
+from .._brownian import AbstractBrownianPath
+from .._custom_types import (
+    AbstractBrownianIncrement,
+    AbstractSpaceTimeLevyArea,
+    AbstractSpaceTimeTimeLevyArea,
+    BoolScalarLike,
+    DenseInfo,
+    FloatScalarLike,
+    IntScalarLike,
+    RealScalarLike,
+    VF,
+    Y,
+)
+from .._local_interpolation import LocalLinearInterpolation
+from .._solution import RESULTS
+from .._term import AbstractTerm, MultiTerm, ODETerm
+from .base import AbstractSolver
+
+
+if TYPE_CHECKING:
+    from typing import ClassVar as AbstractClassVar
+else:
+    from equinox import AbstractClassVar
+
+_ErrorEstimate: TypeAlias = Optional[Y]
+_SolverState: TypeAlias = None
+_CarryType: TypeAlias = tuple[PyTree[Array], PyTree[Array], PyTree[Array]]
+
+
+class AbstractStochasticCoeffs(eqx.Module):
+    a: eqx.AbstractVar[Union[Float[np.ndarray, " s"], tuple[np.ndarray, ...]]]
+    b: eqx.AbstractVar[Union[Float[np.ndarray, " s"], FloatScalarLike]]
+
+    def check(self):
```

Comment:
You probably want an `@abc.abstractmethod` here.

Can you also add a return type and a short docstring, since it looks like this is doing more than just checking invariants?